### PR TITLE
Built PF requests page

### LINKED ISSUE #72: Create Participating Family Request Page
[Project/Issue Tracker](https://github.com/Code-4-Community/lucys-love-bus-frontend/projects/1)

### Objective

As a user I would like to be able to request to be a participating family.
 
### Tech Spec

Relevant API Docs
https://docs.c4cneu.com/llb/llb-api-spec/#-post-api-v1-protected-requests-

- [x] User is able to navigate to a page that allows them to send a request to be a PF
- [x] Page contains and explanation of what this is, when they will be approved, etc. (follow design where possible)
- [x] List of all previous requests
- [x] Shows the current state of the user (GM or PF)
- [ ] User is automatically navigated to this page after setting contacts if query param "pf" is in the URL (for participating family account creation flow)

### Design Artifacts

Current site:

image

### LINKED ISSUE #107: Fix Privilege Level Enum
[Project/Issue Tracker](https://github.com/Code-4-Community/lucys-love-bus-frontend/projects/1)

### Objective

The Enum in the frontend is not consistent with the backend! We need to add the PF type as well.

### PR DESCRIPTION
**Pull request by anirudh723**
Built the Participating Family request page. User can view their previously made requests, and also submit a new request. Information about their role is also displayed. 

## Issue

Closes #72.
Closes #107.

## Changes

- Added personal requests container
- Added components to represent the previous requests + current request
- Added another enum type for PrivilegeLevel -> PF
- Added the necessary routes
- Added redux integration
- Added a new color

## Screenshots

Multiple previous requests (none pending so user can submit a new one):
image

No previous requests (just shows the submit new request part):
image

Modal about the user role:
image

Current request pending (this happens when there's a previous request that is pending):
image

### CODE REVIEW

File: src/components/personalRequests/CurrentRequest.tsx
Comment (rymaju) [inline src/components/personalRequests/CurrentRequest.tsx:66]: we can also call the get requests dispatch instead of reloading right?

File: src/components/personalRequests/PreviousRequests.tsx
Comment (rymaju) [inline src/components/personalRequests/PreviousRequests.tsx:19]: 👀 in general we want to avoid !important. I know you did this because AntD wasnt working though. I would say im "ok" with this, but we should avoid this in the future if possible.

File: src/components/personalRequests/PreviousRequests.tsx
Comment (rymaju) [inline src/components/personalRequests/PreviousRequests.tsx:59]: I believe that when we map JSX elements we need to assign a key to each of the mapped elements

File: src/containers/personalRequests/ducks/selectors.ts
Comment (rymaju) [inline src/containers/personalRequests/ducks/selectors.ts:4]: ⛏️  I think .length > 0 would be more readable than !!

**Review by rymaju — COMMENTED** [review]
Did you make sure you ran npm run check? I think we need to fix that map problem or we'll get an error. Theres a couple of small nitpicks but I think as long as the code passes our checks I would be ok with merging it in.

File: src/containers/personalRequests/ducks/selectors.ts
Comment (rymaju) [reply src/containers/personalRequests/ducks/selectors.ts:4]: We could also use an ormap here:
`return requests.some(request => request.status === 'PENDING')`

**Comment by rymaju** [comment]
Why are github actions not running? 🤔

File: src/components/personalRequests/CurrentRequest.tsx
Comment (anirudh723) [reply src/components/personalRequests/CurrentRequest.tsx:66]: Yeah I think that would work as well

**Review by rymaju — APPROVED** [review]
Awesome, ran npm run check locally and theres no new eslint warnings or typescript errors. I still want to wait on github actions before we merge though.

**Pull request merged**
